### PR TITLE
Add AsciiDoc build infrastructure

### LINKED ISSUE #873: Support building of user documentation from AsciiDoc
As discussed in the recent CDT project call, we should support the building of CDT user documentation from AsciiDoc.

### PR DESCRIPTION
**Pull request by jld01**
Resolves #873

### CODE REVIEW

File: doc/org.eclipse.cdt.doc.user/src/asciidoc/example.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ifdef::env-github[]
+:imagesdir: https://raw.githubusercontent.com/eclipse-cdt/cdt/main/doc/org.eclipse.cdt.doc.user/images
```
Comment (jld01) [inline doc/org.eclipse.cdt.doc.user/src/asciidoc/example.adoc:2]: This is a github-specific directive to locate images in the specified location for the puspose of rendering documents in github itself. Note that new images added by a PR won't yet exist at the required location. Example: https://github.com/jld01/cdt/blob/adoc-example/doc/org.eclipse.cdt.doc.user/src/asciidoc/example.adoc

File: doc/org.eclipse.cdt.doc.user/.asciidoctorconfig.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// 
+// Did not find any configuration files, so creating this at project root level.
+// If you do not like those files to be generated - you can turn it off inside Asciidoctor Editor preferences.
+// 
+// You can define editor specific parts here.
+// For example: with next line you could set imagesdir attribute to subfolder "images" relative to the folder where this config file is located.
+// :imagesdir: {asciidoctorconfigdir}/images
+// 
+// For more information please take a look at https://github.com/de-jcup/eclipse-asciidoctor-editor/wiki/Asciidoctor-configfiles
+:imagesdir: {asciidoctorconfigdir}/images
```
Comment (jld01) [inline doc/org.eclipse.cdt.doc.user/.asciidoctorconfig.adoc:13]: We instruct the AsciiDoc editor to look for images in the `images` folder of the project. Example:
image

**Comment by jld01** [comment]
@jonahgraham, @Torbjorn-Svensson, @jjohnstn and all, following our discussion on the CDT project call this week, this PR is intended to illustrate how we could integrate _AsciiDoc_ content within the existing CDT user documentation plugin for the purpose of adding missing topics and gradually migrating existing topics over time. Comments are welcome.

If we are all happy with this approach, we could start by adding topics relating to _core build_.

**Comment by jonahgraham** [comment]
This looks great - thanks @jld01 

An early target for docs would be the current contents of https://github.com/eclipse-cdt/cdt-lsp/blob/master/README.md - although maybe that should be kept in a (new) org.eclipse.cdt.lsp.docs bundle?

cc: @ghentschke

File: doc/org.eclipse.cdt.doc.user/src/asciidoc/example.adoc [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Contributors:
+    John Dallaway - initial content
+////
+
+// support image rendering and table of contents within GitHub
+ifdef::env-github[]
+:imagesdir: https://raw.githubusercontent.com/eclipse-cdt/cdt/main/doc/org.eclipse.cdt.doc.user/images
+:toc:
+:toc-placement!:
```
Comment (jld01) [inline doc/org.eclipse.cdt.doc.user/src/asciidoc/example.adoc:18]: We need these github-specific directives to support presentation of a table of contents in github.

File: doc/org.eclipse.cdt.doc.user/pom.xml [hunk 4]
```diff
@@ -40,6 +40,32 @@
...
+						</goals>
+						<configuration>
+							<outputDirectory>${project.basedir}/html</outputDirectory>
+							<attributes>
+								<icons>font</icons>
+								<imagesdir>../images</imagesdir>
+								<linkcss />
+								<source-highlighter>coderay</source-highlighter>
+								<stylesdir>..</stylesdir>
+								<stylesheet>help.css</stylesheet>
```
Comment (jld01) [inline doc/org.eclipse.cdt.doc.user/pom.xml:62]: We use the existing CDT help stylesheet

**Comment by jld01** [comment]
@jonahgraham this is all working well now. A few additions to our common stylesheet are required to support Asciidoctor-generated HTML tables and code blocks. Are you OK with the MIT license on the definitions brought in from the default Asciidoctor stylesheet?

If all is well, I propose that we merge everything except the `example.adoc` file.

**Comment by jld01** [comment]
I have received confirmation from @jonahgraham that the MIT-licensed CSS definitions are OK.

**Pull request merged**